### DISCUSSION TITLE
AndTrigger for Cron with Start and End datetime

### DISCUSSION THREAD

**Discussion by jenfredwell** [agronholm/apscheduler#637] [Q&A] [answered]
Hello all!

This is a great library!  Thank you!  I am attempting to set up a job that will run on certain days at a certain time, but has an end date where it will no longer run.  Here is what I am currently trying:

```
    name = 'cron-test'
    tod = time.fromisoformat('14:45:00-06:00')
    start = datetime.fromisoformat('2022-06-14 14:00:00-06:00')
    end = datetime.fromisoformat('2022-06-24 18:00:00-06:00')

    trigger = AndTrigger([CronTrigger(day_of_week='tue,thu',
                                      hour=tod.hour,
                                      minute=tod.minute,
                                      second=tod.second,
                                      timezone=tod.tzinfo),
                          IntervalTrigger(start_date=start,
                                          end_date=end)])
    ap_job = scheduler.add_job(run_pipeline_job,
                               trigger,
                               name=name,
                               args=[name],
                               max_instances=1,  # Only run one job at a time
                               misfire_grace_time=None)  # Allow job to run even if it's very late
    logger.info(f'Added to scheduler DB - {ap_job}')
```

This prints out:
```
INFO:apscheduler.scheduler:Added job "cron-test" to job store "default"
INFO:scheduler:Added to scheduler DB - cron-test (trigger: and[cron[day_of_week='tue,thu', hour='14', minute='45', second='0'], interval[0:00:01]], next run at: 2022-06-14 14:45:00 UTC-06:00)
```

However, at 14:45 (my local time) the job does not run.  I am assuming it has to do with not setting the interval in the `IntervalTrigger` and having it default to `interval[0:00:01]`.  I'm not sure if doing this is possible, but would love to hear any feedback!

Another note, I am using a postgres store and when I start my application back up with this job in the db, the application does not start up completely until I delete this job manually from the db.

I am using `APScheduler==3.9.1`

Thank you!

**Comment by agronholm** [comment, marked as answer]
Is there a reason why you're not specifying start/end times in the cron trigger itself? That would at least be a lot simpler. If you can explain exactly what you're trying to achieve with this trigger, I can probably help.

**Comment by jenfredwell** [comment]
Wow, somehow I had overlooked those arguments for the CronTrigger!  That is MUCH simpler.  Apologies for the dumb question.

**Reply by agronholm** [reply]
No harm done :)
